Sources/Client/Client/Client.swift

```diff
@@ -229,3 +244,17 @@ extension Client {
         }
     }
 }
+
+public struct Subscription {
+    private let onCancel: (String) -> ()
+    let uuid: String
+    
+    init(onCancel: @escaping (String) -> ()) {
+        self.onCancel = onCancel
+        uuid = UUID().uuidString
+    }
+    
+    public func cancel() {
+        onCancel(uuid)
+    }
```

Comment:
```suggestion
    }
    
    deinit() {
        onCancel(uuid)
    }
```

Reply:
The whole purpose of this to _not_ call `onCancel` on `deinit`.....

Reply:
why? It's obvious to do that to avoid leaks from developers.

Reply:
It's super clear and Swift power of memory management.